### DISCUSSION TITLE
how junit run

### DISCUSSION THREAD

**Discussion by Zhouqiee** [junit-team/junit-framework#3260] [Q&A]
java programs have an entry method that executes logic for main() as main()-> execution logic. I wonder how junit becomes junit-> execution logic by replacing main(). Does it encapsulate a main()? Or it is not implemented by a java program, but by a development tool similar to eclipse

**Comment by jbduncan** [comment]
By my understanding, the [command-line version](https://junit.org/junit5/docs/5.9.2/user-guide/#running-tests-console-launcher) of JUnit 5 has its own [`main()` method](https://github.com/junit-team/junit5/blob/ae32d4359de66cd32c76459580b1cc77113ff077/junit-platform-console/src/main/java/org/junit/platform/console/ConsoleLauncher.java#L41) which sets up and starts all the tests.

By comparison, I believe tools like Gradle, Maven, Eclipse and IntelliJ (that is, tools written in JVM programming languages) start JUnit 5 through the [JUnit Platform Launcher API](https://junit.org/junit5/docs/5.9.2/user-guide/#launcher-api).

I'm guessing that tools written in or configured with non-JVM programming languages run the command-line version of JUnit 5 as an external JVM process.

**Comment by sormuras** [comment]
Here's an example how to compile classes (main and test) and run JUnit with only using tools shipping with the JDK and standalone distribution of JUnit.

- https://stackoverflow.com/questions/52373469/how-to-launch-junit-5-platform-from-the-command-line-without-maven-gradle/52373592#52373592
